dmscripts/upload_dos_opportunities_email_list.py

```diff
@@ -0,0 +1,46 @@
+from dmscripts.helpers.supplier_data_helpers import SupplierFrameworkData
+
+
+def find_user_emails(supplier_users, services):
+    """Return email addresses for any supplier who has a service in services."""
+    email_addresses = []
+    for service in services:
+        for user in supplier_users.get(service['supplierId'], []):
+            email_addresses.append(user['email address'])
+    return email_addresses
```

Comment:
:warning:

I couldn't disagree more. Firstly you're disregarding the next person to come along to the code and want to know what this block is up to. Written my way it is *much* easier to see that you're just performing a familiar pattern (and haven't screwed it up) because youre *describing the result you want*.

Also the "tests pass" thing totally depends on how good you are at writing tests, and *nobody* is as good as they think they are.